vm/systemSmartContracts/eei.go

```diff
@@ -380,6 +387,23 @@ func (host *vmContext) AddTxValueToSmartContract(value *big.Int, scAddress []byt
 	destAcc.BalanceDelta = big.NewInt(0).Add(destAcc.BalanceDelta, value)
 }
 
+// IsValidator returns true if the validator is in eligible or waiting list
+func (host *vmContext) IsValidator(blsKey []byte) bool {
+	acc, err := host.validatorAccountsDB.GetExistingAccount(blsKey)
+	if err != nil {
+		return false
+	}
+
+	validatorAccount, ok := acc.(state.PeerAccountHandler)
+	if !ok {
+		return false
+	}
+
+	isValidator := validatorAccount.GetList() == string(core.EligibleList) ||
```

Comment:
This is so wrongly named (validatorAccount.GetList())

Reply:
added TODO.